src/app/profile/components/profile/profile.component.ts

```diff
@@ -254,4 +254,26 @@ export class ProfileComponent extends BaseAuthenticatedRoutableComponent impleme
 			},
 		);
 	}
+
+    async delete() {
+		if (
+			await this.dialogService.confirmDialog.openTrueFalseDialog({
+				dialogTitle: this.translate.instant('Profile.deleteAccount'),
+				dialogBody: this.translate.instant('Profile.deleteAccountConfirm'),
+				resolveButtonLabel: this.translate.instant('Profile.deleteAccountYes'),
+				rejectButtonLabel: this.translate.instant('General.cancel'),
+			})
+		) {
+            this.profile.delete().then(
+                () => {
+                    this.sessionService.logout();
+                    // Not sure why I need the timeout, but
+                    // otherwise the message isn't shown
+                    setTimeout(() => this.messageService.reportMajorSuccess(this.translate.instant('Profile.deleteAccountSuccess'), true));
+                    this.router.navigate(['/public/start']);
+                },
+                (error) => this.messageService.reportHandledError(this.translate.instant('Profile.deleteAccountFailure') + ` ${error.response._body}`)
```

Comment:
Wouldnt you need the timeout here as well then?

Reply:
Actually it's not needed here because there's no redirect. I improved the reporting anyway.